src/core/src/core_logic/RebootManager.py

```diff
@@ -14,116 +14,167 @@
             return False
 
-        # attempt to reboot is enough time is available
-        if self.reboot_setting == Constants.REBOOT_ALWAYS or (self.reboot_setting == Constants.REBOOT_IF_REQUIRED and reboot_pending):
+        # Try to reboot - if enough time is available
+        if self.__reboot_setting == Constants.REBOOT_ALWAYS or (self.__reboot_setting == Constants.REBOOT_IF_REQUIRED and reboot_pending):
             if self.is_reboot_time_available(current_time_available):
-                self.composite_logger.log(' - Reboot is being scheduled, as per patch installation configuration (' + str(self.reboot_setting) + ').')
-                self.composite_logger.log(" - Reboot-pending status: " + str(reboot_pending))
-                self.start_reboot()
+                self.composite_logger.log_debug('[RM] Reboot is being scheduled, as per customer configuration ({0}). [RebootPending={1}][CurrentTimeAvailable={2}]'.format(str(self.__reboot_setting), str(reboot_pending), str(current_time_available)))
+                self.__start_reboot(maintenance_window_available_time_in_minutes=current_time_available)
                 return True
             else:
-                error_msg = ' - There is not enough time to schedule a reboot as per patch installation configuration (' + str(self.reboot_setting) + '). Reboot-pending status: ' + str(reboot_pending)
+                # Maintenance window will be marked exceeded as reboot is required and not enough time is available
+                error_msg = '[RM][!] Insufficient time to schedule a required reboot ({0}). [RebootPending={1}][CurrentTimeAvailable={2}]'.format(str(self.__reboot_setting), str(reboot_pending), str(current_time_available))
                 self.composite_logger.log_error(error_msg)
-                self.status_handler.add_error_to_status("Reboot Management" + str(error_msg), Constants.PatchOperationErrorCodes.DEFAULT_ERROR)
-                self.maintenance_window_exceeded_flag = True
+                self.status_handler.add_error_to_status(str(error_msg), Constants.PatchOperationErrorCodes.DEFAULT_ERROR)
+                self.__maintenance_window_exceeded_flag = True
                 return False
 
-    def is_reboot_pending(self):
-        return self.package_manager.force_reboot or (self.status_handler and self.status_handler.is_reboot_pending)
+        # No-op - This code should never be reached. If seen, it indicates a bug in the code.
+        self.composite_logger.log_error('[RM] Bug-check: Unexpected code branch reached. [RebootSetting={0}][RebootPending={1}]'.format(str(self.__reboot_setting), str(reboot_pending)))
+        return False
+
+    def __start_reboot(self, message="Azure VM Guest Patching initiated a reboot after an 'InstallPatches' operation.", maintenance_window_available_time_in_minutes=0):
```

Comment:
Not necessarily "after an 'InstallPatches' operation". Can happen at the start or in between too

Reply:
Good callout; will adjust the text.

Reply:
Text is now: "Azure VM Guest Patching initiated a reboot as part of an 'InstallPatches' operation."